### PR TITLE
3-GET /api completed

### PR DESCRIPTION
**Pull request by Kingsl8yer**
3-GET /api completed with tests as well

### CODE REVIEW

File: MVC/controllers/topicsController.js [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+    .then((topics) => {
+      res.status(200).send({ topics });
+    })
+    .catch(next);
+};
+
+exports.getAllEndpoints = (req, res, next) => {
+  res.status(200).send({
+    endpoints: {
+      "GET /api/topics": {
```
Comment (Augs0) [inline MVC/controllers/topicsController.js]: This is fine for now but moving forward you will find you have to go and update this every time. Remember, we already have this data somewhere. What could we send instead?

File: __tests__/index.test.js [hunk 2]
```diff
@@ -31,3 +31,36 @@ describe("GET /api/topics", () => {
...
   });
 });
+
+describe("GET /api", () => {
+  test("status 200: responds with an object containing all endpoints", () => {
+    return request(app)
+      .get("/api")
+      .expect(200)
+      .then(({ body }) => {
+        for (let key in body.endpoints) {
```
Comment (Augs0) [inline __tests__/index.test.js]: Before any iteration in a test, we should check the array's length to be sure that we have something to iterate over

File: __tests__/index.test.js [hunk 3]
```diff
@@ -31,3 +31,36 @@ describe("GET /api/topics", () => {
...
+          expect(typeof body.endpoints[key].description).toBe("string");
```
Comment (Augs0) [inline __tests__/index.test.js]: This is very thorough and valid, but an easier way might have been to use a .toEqual against the file we are using with the endpoints

**Review by Augs0 — APPROVED** [review]
Well done, Miguel. You have achieved the objectives of the task and can merge, but there are a few things I've mentioned here that might improve the code here that you could implement if you wish. Most importantly, I would probably change what you are sending in the controller to prevent more work on future tasks.

File: __tests__/index.test.js (hunk 3 above)
Comment (Kingsl8yer) [reply __tests__/index.test.js]: Thank you, now I got it, it seems I didn't understand quite well the task!

**Pull request merged**